### PR TITLE
[18.0][MIG] document_knowledge: Migration to 18.0

### PR DESCRIPTION
**Pull request by Kimkhoi3010**

### CODE REVIEW

**Review by jdoutreloux — COMMENTED** [review]
It might be nice to set the `path` attribute on the `knowledge_action_documents` to have a user-friendly url. 
But otherwise lgtm

File: document_knowledge/views/document_knowledge.xml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="knowledge_action_documents" model="ir.actions.act_window">
+        <field name="name">Documents</field>
```
Comment (jdoutreloux) [inline document_knowledge/views/document_knowledge.xml:4]: `<field name="path">documents</field>`

Comment (Kimkhoi3010) [reply document_knowledge/views/document_knowledge.xml:4]: @jdoutreloux thanks for your time!!

**Review by TDu — APPROVED** [review]

**Review by jdoutreloux — APPROVED** [review]

**Review by houssine78 — APPROVED** [review]
LGTM

**Comment by houssine78** [comment]
@pedrobaeza a new PR that can be merge here :)

File: document_knowledge/models/res_config.py [hunk 2]
```diff
@@ -9,25 +9,17 @@ class DocumentKnowledgeConfigSettings(models.TransientModel):
 
     module_attachment_indexation = fields.Boolean(
         "Attachments List and Document Indexation",
-        help="Document indexation, full text search of attachements.\n"
```
Comment (pedrobaeza) [inline document_knowledge/models/res_config.py]: I prefer that you remove all the options, as this is not valid if the rest of the modules aren't there.

File: document_knowledge/readme/CREDITS.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
 The development of this module has been financially supported by:
 
 - Camptocamp
+
+The migration of this module from 17.0 to 18.0 was financially supported
```
Comment (pedrobaeza) [inline document_knowledge/readme/CREDITS.md]: Remove the previous text.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Controversial opinion about removing the settings part, but this is a frequent source of questions/problems cc @max3903

File: document_knowledge/models/res_config.py (hunk 2 above)
Comment (Kimkhoi3010) [reply document_knowledge/models/res_config.py]: @pedrobaeza Thanks for your time!!
Can you explain more so I can better understand what you want.

Comment (pedrobaeza) [reply document_knowledge/models/res_config.py]: I mean that having these `module_*` fields is for installing modules that are still not migrated, but even if migrated, you may not have those modules present on your Odoo instance, so clicking on the checks provoke confusion and misleading in users, so I propose to remove these fields from the configuration.

**Comment by pedrobaeza** [comment]
Thanks for the changes. I think you have stripped now too much, as I was just talking about `module_*` fields, not the rest.

**Comment by Kimkhoi3010** [comment]
> Thanks for the changes. I think you have stripped now too much, as I was just talking about `module_*` fields, not the rest.

It was my fault. I fixed it. Thanks for your time!!

**Comment by houssine78** [comment]
thanks @pedrobaeza for this insightful review.

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**